test/ex_rlp/property_test.exs

```diff
@@ -0,0 +1,44 @@
+defmodule ExRLP.PropTest do
+  @moduledoc false
+
+  use PropCheck
+  use ExUnit.Case
+
+  def safe_decode(t) do
+    try do
+      ExRLP.decode(t)
+    rescue
+      _ -> false
+    end
+  end
+
+  property "decoding is one to one", [1000, :verbose, max_size: 100, constraint_tries: 100_000] do
+    gen = such_that(bin <- binary(), when: false != safe_decode(bin))
+
+    forall l <- list(gen) do
+      mapsto = for item <- :lists.usort(l), do: ExRLP.decode(item)
+      :lists.sort(mapsto) == :lists.usort(mapsto)
```

Comment:
I'm not sure what this does, can you throw in some comments/expand the description of the test/make the assertion more explicit, without the use of `usort`s? (ideally some mix of these 3).

I suspect that were checking that two sets are the same, but not sure how this proves the "one to one"...

Reply:
Done, please see again :-)

Reply:
On one-to-one proof. It looks for two binaries that violate one-to-one property of decoder. It is obviously not a proof since it is not exhaustive. Also it is a mapping in one direction only and no claims about the state of encoder are made in this particular test.

Reply:
perfect, thanks for dumbing it down ;). I'm wondering; how sure are we, that https://github.com/mana-ethereum/ex_rlp/pull/24/files#diff-0dbbe73b6149be5f2b00785f37f44760R27 doesn't yield a very small testing set? I'm assuming it doesn't and probably `propcheck` is smart enough to not feed us duplicates, then in such case maybe let's put a sanity precondition check there, that `usort==sort` (I can do it later, if I finally get to taking this over).

> On one-to-one proof

ah, sorry I used "proves" too liberally, as in "attempts to prove a point/verify a property". All is understood